src/main/java/carpet/patches/TickSyncedBorderExtent.java

```diff
@@ -0,0 +1,160 @@
+package carpet.patches;
+
+import carpet.CarpetServer;
+import carpet.helpers.TickSpeed;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.Mth;
+import net.minecraft.world.level.border.BorderChangeListener;
+import net.minecraft.world.level.border.BorderStatus;
+import net.minecraft.world.level.border.WorldBorder;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import java.util.Arrays;
+import java.util.OptionalDouble;
+
+/**
+ * This class is essentially a copy of {@link net.minecraft.world.level.border.WorldBorder.MovingBorderExtent}
+ * but instead of using real time to lerp the border
+ * this class uses the in game ticks.
+ */
+@SuppressWarnings("JavadocReference")
+public class TickSyncedBorderExtent implements WorldBorder.BorderExtent
+{
+	private final WorldBorder border;
+	private final long realDuration;
+	private final double tickDuration;
+	private final double from;
+	private final double to;
+
+	private int ticks;
+
+	public TickSyncedBorderExtent(WorldBorder border, long realDuration, double from, double to)
+	{
+		this.border = border;
+		this.realDuration = realDuration;
+		this.tickDuration = realDuration / 50.0;
+		this.from = from;
+		this.to = to;
+		this.ticks = 0;
+	}
+
+	@Override
+	public double getMinX()
+	{
+		int maxSize = this.border.getAbsoluteMaxSize();
+		return Mth.clamp(this.border.getCenterX() - this.getSize() / 2.0, -maxSize, maxSize);
+	}
+
+	@Override
+	public double getMaxX()
+	{
+		int maxSize = this.border.getAbsoluteMaxSize();
+		return Mth.clamp(this.border.getCenterX() + this.getSize() / 2.0, -maxSize, maxSize);
+	}
+
+	@Override
+	public double getMinZ()
+	{
+		int maxSize = this.border.getAbsoluteMaxSize();
+		return Mth.clamp(this.border.getCenterZ() - this.getSize() / 2.0, -maxSize, maxSize);
+	}
+
+	@Override
+	public double getMaxZ()
+	{
+		int maxSize = this.border.getAbsoluteMaxSize();
+		return Mth.clamp(this.border.getCenterZ() + this.getSize() / 2.0, -maxSize, maxSize);
+	}
+
+	@Override
+	public double getSize()
+	{
+		double progress = this.ticks / this.tickDuration;
+		return progress < 1.0 ? Mth.lerp(progress, this.from, this.to) : this.to;
+	}
+
+	@Override
+	public double getLerpSpeed()
+	{
+		return Math.abs(this.from - this.to) / this.tickDuration;
```

Comment:
Isn't this in seconds? (or blocks/second or ms or whatever)

Reply:
Yes, you are correct, this is a mistake - I fixed this earlier for my UHC mod impelmentation but forgot to update it here.